include/zephyr/posix/sys/utsname.h

```diff
@@ -6,15 +6,23 @@
 #ifndef ZEPHYR_INCLUDE_POSIX_SYS_UTSNAME_H_
 #define ZEPHYR_INCLUDE_POSIX_SYS_UTSNAME_H_
 
+#include <zephyr/sys/util_macro.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/* These are for compatibility / practicality */
+#define _UTSNAME_NODENAME_LENGTH                                                                   \
+	COND_CODE_1(CONFIG_POSIX_SINGLE_PROCESS, (CONFIG_POSIX_UNAME_VERSION_LEN), (0))
+#define _UTSNAME_VERSION_LENGTH                                                                    \
+	COND_CODE_1(CONFIG_POSIX_SINGLE_PROCESS, (CONFIG_POSIX_UNAME_VERSION_LEN), (0))
+
 struct utsname {
 	char sysname[sizeof("Zephyr")];
-	char nodename[CONFIG_POSIX_UNAME_NODENAME_LEN + 1];
+	char nodename[_UTSNAME_NODENAME_LENGTH + 1];
```

Comment:
Probably can move the `+ 1` into the macro above as well?

Reply:
I'm kind of leaning toward making this PR as much of a straight text replacement as possible.

There are quite a few other minefields that need to be carefully traversed in terms of standard feature test macros, so I would rather not make drive-by changes if possible.